### ISSUE TITLE
Connecting to MongoDb in Docker throws uncaught promise (in wsl2)

### ISSUE THREAD

**Issue by RobertFent** [denodrivers/deno_mongo#200]
I'm trying to connect the MongoClient to a MongDb container running in Docker.<br>
The docker container is running in wsl2 with the wsl ip 172.18.96.38 and exposing port 27017.

Connecting to it with the following code leads to an error:
```ts
import { MongoClient } from "https://deno.land/x/mongo@v0.23.0/mod.ts";

const uri = 'mongodb://admin:admin@172.18.96.38:27017'
this.mClient = new MongoClient();
await this.mClient.connect(uri);
```
```
error: Uncaught (in promise) Error: MongoError: "Connection failed: MongoError: {\"ok\":0,\"errmsg\":\"no such command: 'hello'\",\"code\":59,\"codeName\":\"CommandNotFound\"}"
      throw new MongoError(`Connection failed: ${e.message || e}`);
            ^
    at MongoClient.connect (https://deno.land/x/mongo@v0.23.0/src/client.ts:26:13)
    at async Function.initMongoClient (file:///home/robert/repos/myDenoWebsite/backend/src/utils/mongoClientWrapper.ts:54:9)
```
According to the errmsg the error is 'no such command: 'hello'.
What does this mean?

This is my docker-compose.yml for mongo btw:
```yml
version: '3.8'

services:
  mongodb:
    image: mongo:4.4-bionic
    container_name: my-mongodb
    ports:
      - 27017:27017
    environment:
      - MONGO_INITDB_ROOT_USERNAME=admin
      - MONGO_INITDB_ROOT_PASSWORD=admin
    volumes:
      - mongodb:/data/db
      - mongoconfig:/data/configdb

volumes:
  mongodb:
  mongoconfig:
```

**Comment by vincenzopalazzo** [comment]
Are your sure that this URL `const uri = 'mongodb://admin:admin@172.18.96.38:27017'` It is a valid URL in the docker compose network?

I think you are missing a `networks: host`.

**Comment by RobertFent** [comment]
@vincenzopalazzo This URL is valid. I can connect to the db via MongoDBAtlas

**Comment by RobertFent** [comment]
> Are your sure that this URL `const uri = 'mongodb://admin:admin@172.18.96.38:27017'` It is a valid URL in the docker compose network?
> 
> I think you are missing a `networks: host`.

@vincenzopalazzo This URL is valid. I can connect to the db via MongoDBAtlas
Using ```network: host``` does not work in wsl as expected

**Comment by vincenzopalazzo** [comment]
>I can connect to the db via MongoDBAtlas

This works because you expose the port 27017, but if the request if inside the from a docker image to another docker image, the network of docker it is not `172.18.96.38`. If you change the expose port to 27018 your MongoDBAtlas will not work anymore.

From the docker documentation

>By default Compose sets up a single network for your app. Each container for a service joins the default network and is both reachable by other containers on that network, and discoverable by them at a hostname identical to the container name.

**Comment by mattijauhiainen** [comment]
I think I broke this in my earlier MR. The `hello` command is not implemented in 4.4.0 which seems to be the version that comes with 4.4-bionic. You could work around by using another base image with mongo 4.4.2 @RobertFent .
